src/components/Camera.ts

```diff
@@ -0,0 +1,296 @@
+                    this.takePicture,
+                    "widget-camera-picture"
+                ), this.createSwitchCameraButton()
+            )
+        );
+    }
+
+    private renderPhoto(): ReactElement<{}> {
+        return createElement("div", { className: classNames("widget-camera-parent") },
+            createElement("img", {
+                alt: "Image could not be found!",
+                src: this.state.screenshot,
+                style: this.bindDimensions()
+            }),
+            this.createContainer(
+                { width: this.pictureWidth },
+                this.createButton(
+                    this.props.recaptureButtonName,
+                    this.props.captureButtonIcon,
+                    this.retakePicture,
+                    "widget-camera-picture"
+                ),
+                this.createButton(
+                    this.props.usePictureButtonName,
+                    this.props.usePictureButtonIcon,
+                    this.onClick,
+                    "widget-camera-switch-button"
+                )
+            )
+        );
+    }
+
+    private createIcon(buttonLabel: string, glyphiconName: string): ReactElement<{}> {
+        return (this.props.captionsToUse === "icons")
+            ? createElement("span", { className: classNames(`glyphicon glyphicon-${glyphiconName}`) })
+            : createElement("button", { className: classNames("btn btn-inverse active") }, buttonLabel);
+    }
+
+    private createButton(buttonLabel: string, glyphiconName: string, action: () => void, buttonclass: string): ReactElement<{}> {
+        return createElement("span", {
+            className: classNames(buttonclass),
+            onClick: action
+        },
+            this.createIcon(buttonLabel, glyphiconName)
+        );
+    }
+
+    private createSwitchCameraButton(): ReactElement<{}> {
+        if (this.availableDevices.length > 1) {
+            return this.createButton(
+                "Switch",
+                this.props.switchCameraIcon,
+                this.changeCamera,
+                "widget-camera-switch-button"
+            );
+        }
+
+        return createElement("div", {});
+    }
+
+    private changeCamera() {
+        const cameraDevicePosition: number = this.state.cameraDevicePosition < (this.availableDevices.length - 1)
+            ? this.state.cameraDevicePosition + 1
+            : 0;
+        this.setState({ cameraDevicePosition, swapCamera: true });
+    }
+
+    private createContainer(property?: object, childOne?: ReactElement<{}>, childTwo?: ReactElement<{}>): ReactElement<{}> {
+        return createElement("div", { style: property }, childOne, childTwo);
+    }
+
+    private renderAlert(message: string): ReactElement<AlertProps> {
+        return createElement(Alert, {
+            bootstrapStyle: "danger",
+            className: "",
+            message
+        });
+    }
+
+    componentWillMount() {
```

Comment:
Please organise your functions properly.
Ideally, the should be as follows:

1. static properties
2. Class instance variable
3. Public functions. For lifecycle methods, arrange in the order in which they'll be called.
4. Private functions. Arrange in a way that eases reading the flow of your code. Don't have a function called in the constructor as the last one in your code.

Reply:
Done